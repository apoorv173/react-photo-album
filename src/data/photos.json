[
  {
    "photoId": 1,
    "photoTitle": "quidem molestiae enim",
    "thumbnailUrl": "/fff"
  },
  {
    "photoId": 2,
    "photoTitle": "sunt qui excepturi placeat culpa",
    "thumbnailUrl": "/111"
  }
]