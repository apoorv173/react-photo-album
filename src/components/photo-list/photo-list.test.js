import React from "react";
import PhotoList from "./photo-list";
import { shallow, mount } from "enzyme";
import photos from '../../data/photos.json';

describe("Photo List Component", () => {

    const onPhotoClick = jest.fn();

    it("Should render photos with user data", () => {
        const tree = shallow(<PhotoList photos={photos} />)
        expect(tree).toMatchSnapshot();
    });

    it('should render all the components', () => {
        const wrapper = mount(<PhotoList photos={photos} onPhotoClick={onPhotoClick} />);
        expect(wrapper.find('.row').length).toBe(1);
        expect(wrapper.find('.thumbnail').length).toBe(2);
        expect(wrapper.find('.thumbnail-img').length).toBe(2);
        expect(wrapper.find('.thumbnail-heading').length).toBe(2);
        expect(wrapper.find('.layer').length).toBe(2);
    });

    it('should call the on click event to load the modal', () => {
        const wrapper = mount(<PhotoList photos={photos} onPhotoClick={onPhotoClick} />);
        expect(wrapper.find('.row').length).toBe(1);
        wrapper.find(".row .thumbnail-wrapper").at(0).simulate("click");
        expect(onPhotoClick).toHaveBeenCalled();
    });
});