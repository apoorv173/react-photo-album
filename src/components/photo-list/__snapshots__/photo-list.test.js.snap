// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo List Component Should render albums with user data 1`] = `ShallowWrapper {}`;

exports[`Photo List Component Should render photos with user data 1`] = `ShallowWrapper {}`;
